testData/solutions/win32.json5

```diff
@@ -3724,6 +3724,263 @@
+    },
+    "com.microsoft.windows.desktopBackgroundColor": {
+        "name": "Windows desktop background personalization",
+        "contexts": {
+            "OS": [
+                {
+                    "id": "win32",
+                    "version": ">=5.0"
+                }
+            ]
+        },
+        "settingsHandlers": {
+            "configureImage": {
+                "type": "gpii.windows.spiSettingsHandler",
+                "liveness": "live",
+                "options": {
+                    "getAction": "SPI_GETDESKWALLPAPER",
+                    "setAction": "SPI_SETDESKWALLPAPER",
+                    "uiParam": 260,
+                    "pvParam": {
+                        "type": "array",
+                        "valueType": "TCHAR",
+                        "length": 260
+                    },
+                    "verifySettings": false
+                },
+                "supportedSettings": {
+                    "ImageConfig": {
+                        "schema": {
+                            "title": "Image",
+                            "description": "The path to the image to be set as the new desktop wallpaper",
+                            "type": "String",
+                            "default": ""
+                        }
+                    }
+                },
+                "capabilitiesTransformations": {
+                    "ImageConfig": {
+                        "transform": {
+                            "type": "fluid.transforms.literalValue",
+                            "input": "",
+                            "outputPath": "value"
+                        },
+                        "path": {
+                            "transform": {
+                                "type": "fluid.transforms.literalValue",
+                                "input": "pvParam"
+                            }
+                        }
+                    }
+                },
+                "inverseCapabilitiesTransformations": {
+                    "http://registry\\.gpii\\.net/applications/com\\.microsoft\\.windows\\.desktopBackgroundColor.Image": "ImageConfig.value"
+                }
+            },
+            "configureSolidColor": {
+                "type": "gpii.windows.nativeSettingsHandler",
+                "liveness": "live",
+                "options": {
+                    "functionName": "SolidColor",
+                    "verifySettings": true
+                },
+                "supportedSettings": {
+                    "SolidColorConfig": {
+                        "schema": {
+                           "title": "Desktop background payload",
+                            "description": "The values that user want to use for that particular background type",
+                            "type": "Object",
+                            "default": {
+                                "r": 0, "g": 0, "b": 0
+                            },
+                            "min": {
```

Comment:
I don't believe this use of "min" applied to non-numeric types is valid - @the-t-in-rtf , could you check over? I imagine that the min/max needs to be applied to each field separately.

Reply:
For starters, "min" and "max" are no longer allowed.  It's "minimum" and "maximum" now.  Also the syntax used is shall  we say overly optimistic about how smart JSON Schema is.  As Antranig said, you have to spell out the min and max for each property of the object, as in:

```
{
  "type": "Object",
  "properties": {
    "additionalProperties": false,
    "r": {
      "minimum": 0,
      "maximum": 255,
      "default": 0,
      "required": true
     },
    "g": {
      "minimum": 0,
      "maximum": 255,
      "default": 0,
      "required": true
     },
    "b": {
      "minimum": 0,
      "maximum": 255,
      "default": 0,
      "required": true
     },
  }
}
``` 

Note that I added a few other key bits.  Each property is required, so you can't omit it.  I've also said that no other properties are allowed.  This is much tighter than the original contract, which would have allowed any object, including an empty one, or one filled with whatever people felt like.  Happy to explain further if the example doesn't clear things up.

Reply:
Note also the typed properties.  The previous contract would have allowed you to stuff anything at all into the "r", "g", and "b" properties as well.

Reply:
Yes, great I will use the provided schema, it looks really coherent with the previous usages.